presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java

```diff
@@ -71,30 +72,21 @@ public class OrcBatchPageSource
 
     private final RuntimeStats runtimeStats;
 
-    private final List<Boolean> isRowPositionList;
+    private final boolean[] isRowNumberList;
 
     private final RowIDCoercer coercer;
 
-    public OrcBatchPageSource(
-            OrcBatchRecordReader recordReader,
-            OrcDataSource orcDataSource,
-            List<HiveColumnHandle> columns,
-            TypeManager typeManager,
-            OrcAggregatedMemoryContext systemMemoryContext,
-            FileFormatDataSourceStats stats,
-            RuntimeStats runtimeStats,
-            byte[] rowIDPartitionComponent,
-            String rowGroupId)
-    {
-        this(recordReader, orcDataSource, columns, typeManager, systemMemoryContext, stats, runtimeStats, nCopies(columns.size(), false), rowIDPartitionComponent, rowGroupId);
-    }
-
     /**
      * @param columns an ordered list of the fields to read
-     * @param isRowPositionList list of indices of columns. If true, then the column with the matching
-     *     index is a row number. If false, it isn't. This should have the same length as {@code columns},
-     *     and the columns in each list should match up.
+     * @param isRowNumberList list of indices of columns. If true, then the column then the column
+     *     at the same position in {@code columns} is a row number. If false, it isn't.
+     *     This should have the same length as {@code columns}.
+     * #throws IllegalArgumentException if columns and isRowNumberList do not have the same size
      */
+    // TODO(elharo) I notice that the columns and isRowNumberList match is an antipattern I used
+    // to take points off for on student projects for bad OO design. Most likely HiveColumnHandle
+    // should know whether it's a row number or not. Alternately, we could define a class
+    // that includes both a column handle and the row number boolean.
```

Comment:
```suggestion
    // TODO: make HiveColumnHandle contain whether it's a row number or not. Alternately, define a class
    // that includes both a column handle and the row number boolean.
```
Please remove extraneous commentary so that it's very clear what needs to be done.

Reply:
done